crates/router/src/db/merchant_account.rs

```diff
@@ -75,12 +76,29 @@ impl MerchantAccountInterface for Store {
         merchant_key_store: &domain::MerchantKeyStore,
     ) -> CustomResult<domain::MerchantAccount, errors::StorageError> {
         let conn = connection::pg_connection_write(self).await?;
-        merchant_account
+
+        let key_store = merchant_key_store
+            .clone()
```

Comment:
Why do you need a clone here?

Reply:
merchant_key_store is used in line 104 as well

Reply:
You can just clone the key and use it in 104